src/bin/cargo/commands/fix.rs

```diff
@@ -71,8 +71,7 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
     // Unlike other commands default `cargo fix` to all targets to fix as much
     // code as we can.
     let root_manifest = args.root_manifest(gctx)?;
-    let mut ws = Workspace::new(&root_manifest, gctx)?;
-    ws.set_resolve_honors_rust_version(args.honor_rust_version());
+    let ws = args.workspace(gctx)?;
```

Comment:
Using `args.workspace()` here have `cargo fix` to respect [`-Zavoid-dev-deps`](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html?highlight=avoid-dev-d#avoid-dev-deps), which previous it didnt.

I think it is intentionally, as `cargo fix` by default also builds test targets, which requires dev-dependencies.